source/Nuke.GlobalTool/Program.Fix.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Copyright 2019 Maintainers of NUKE.
+// Distributed under the MIT License.
+// https://github.com/nuke-build/nuke/blob/master/LICENSE
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using JetBrains.Annotations;
+using Nuke.Common;
+using Nuke.Common.IO;
+using Nuke.Common.ProjectModel;
+using Nuke.Common.Tooling;
+using Nuke.Common.Utilities.Collections;
+using static Nuke.Common.Utilities.ConsoleUtility;
+
+namespace Nuke.GlobalTool
+{
+    partial class Program
+    {
+        private const string VERSION_MANUAL = "manual";
+        private const string VERSION_NONE = "none";
+
+        private const string PACKAGETYPE_DOWNLOAD = "PackageDownload";
+        private const string PACKAGETYPE_REFERENCE = "PackageReference";
+
+        [UsedImplicitly]
+        private static int Fix(string[] args, [CanBeNull] string rootDirectory, [CanBeNull] string buildScript)
+        {
+            if (rootDirectory == null)
+            {
+                // TODO: parse --root parameter
+                return 0;
+            }
+
+            var missingPackageFile = Constants.GetMissingPackageFile((AbsolutePath)rootDirectory);
+            if (!File.Exists(missingPackageFile))
+            {
+                Logger.Success("Nothing to fix, all good!");
+                return 0;
+            }
+
+            var missingPackageContents = File.ReadAllLines(missingPackageFile);
+            var buildProjectFilePath = ((AbsolutePath)missingPackageContents.FirstOrDefault())
+                .NotNull("Invalid build project file reference");
+
+            // TODO: better way?
+            if (IsProjectTypeValidForFix(buildProjectFilePath.Parent))
+            {
+                Logger.Error("Only .NET Core projects are supported.");
+                return 0;
+            }
+
+            var project = ProjectModelTasks.ParseProject(buildProjectFilePath).NotNull();
+            var packages = missingPackageContents
+                .Skip(1)
+                .Where(x => Regex.IsMatch(x, "^[^@]+@.+$"))
```

Comment:
I'm not sure what this is :)

Reply:
rewrote it to make clearer what's happening.